src/core/dbus-cgroup.c

```diff
@@ -570,6 +602,85 @@ static int bus_cgroup_set_transient_property(
                         }
                 }
 
+                return 1;
+        } else if (streq(name, "BPFProgram")) {
+                size_t n = 0;
+
+                r = sd_bus_message_enter_container(message, 'a', "s");
```

Comment:
as above use `(ss)` as array type

Reply:
Changed and tested with `mkosi` image:
```
[root@image ~]# SYSTEMD_SECCOMP=0 systemd-nspawn -D/ --volatile=yes --capability=cap_net_admin  --bind-ro=/sys:/sys --property=BPFProgram=egress:/sys/fs/bpf/trivial/cgroup_skb_egress -- bpftool c t /sys/fs/cgroup/machine.slice
Spawning container image on /.
Press ^] three times within 1s to kill container.
CgroupPath
ID       AttachType      AttachFlags     Name           
/sys/fs/cgroup/machine.slice/machine-image.scope
    239      egress          multi           trivial_egress 
    249      device          multi                          
Container image exited successfully.
[root@image ~]# 
```
Also since the property is array type updated `bus_append_cgroup_property` and `systemctl show`:
```
[root@image ~]# systemctl show bpf-foreign -p BPFProgram
BPFProgram=egress:/sys/fs/bpf/trivial/cgroup_skb_egress
[root@image ~]# 
```